sw/host/opentitanlib/src/image/image.rs

```diff
@@ -83,7 +83,12 @@ impl SigverifyParams {
     // Verify the optional SPX+ signature.
     pub fn spx_verify(&self, b: &[u8], domain: SpxDomain) -> Result<()> {
         if let Some(spx) = &self.spx_sig_params {
-            spx.key.verify(domain, &spx.signature, b)?;
+            let hash = sha256::sha256(b).to_le_bytes();
```

Comment:
The hash calculation should be inside the match case for `PreHashedSha256`.

You probably have to use a `Cow` (copy-on-write) reference to deal with the difference between the hash result (`Vec<u8>`) vs. the slice (`[u8]`):
```
let msg = match domain {
  PreHashedSha256 => Cow::from(sha256::sha256(b).to_le_bytes()),
  _ => Cow::from(b),
}
```

Reply:
I tried to create the hash in the match case, but it did not have long enough life time, looks like `Cow::from()` takes care of this, done.